src/Drupal/UpdateKernel.php

```diff
@@ -0,0 +1,12 @@
+<?php
+namespace Drush\Drupal;
+
+use Drupal\Core\Update\UpdateKernel as DrupalUpdateKernel;
+
+/**
+ * Overridden version of UpdateKernel adapted to the needs of Drush.
+ */
+class UpdateKernel extends DrupalUpdateKernel
```

Comment:
We should probably just use `Drupal\Core\Update\UpdateKernel` directly for the updatedb command. This means that commands that bootstrap with the UpdateKernel cannot use Drush commands. If we are okay with this limitation, then we wouldn't need the DrupalKernelTrait.

The one way in which this could go really wrong would be if updatedb did a cache rebuild at the end (in the Drupal code), and that cache rebuild cached all of the services, excluding the Drush services. We could get around this by using `drush_invoke_process` at the end of updatedb (in Drush code) to call the cache:rebuild command. It's likely that we already do this.

Reply:
Already reverted to the vanilla `UpdateKernel`. Still need to see where this will go wrong.

Reply:
It appears we still need those overrides. `drush_invoke_process()` relies on the `DrushCommand` services being available on the container. They get added by the `DrushServiceModifier`.